packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -114,9 +111,17 @@ export default class ChildProcessWorker implements WorkerInterface {
       this._child.kill('SIGKILL');
     }
 
-    this._state = WorkerStates.STARTING;
+    this.state = WorkerStates.STARTING;
 
     const forceColor = stdoutSupportsColor ? {FORCE_COLOR: '1'} : {};
+    const silent = this._options.silent ?? true;
+
+    if (!silent) {
+      console.warn('Unable to detect out of memory event if silent === false');
```

Comment:
we should only log this if `_childIdleMemoryUsageLimit` is passed I think? Then we can probably also make it an error instead of warning

Reply:
same comment applies to thread worker - shouldn't we only monitor for OOM if the option is passed, otherwise keep old behavior?

Reply:
> we should only log this if `_childIdleMemoryUsageLimit` is passed I think? Then we can probably also make it an error instead of warning

Yes to the first bit, i'm inclined to say no to the second. It can be useful for debugging to see the child output when a memory limit is in place, for example to see the exact thrown error message.

Reply:
> same comment applies to thread worker - shouldn't we only monitor for OOM if the option is passed, otherwise keep old behavior?

Actually, ignore my previous comment....

Detecting an out of memory event and the worker crashing is independent of monitoring of the idle memory usage and rebooting as necessary. The `_childIdleMemoryUsageLimit` applies to the latter case, the former is building and improving on #13054

Reply:
I've added some comments to explain